common/src/main/java/io/netty5/util/internal/PlatformDependent0.java

```diff
@@ -811,29 +779,19 @@ static int hashCodeAsciiSanitize(byte value) {
         return value & 0x1f;
     }
 
+    @Deprecated
     static ClassLoader getClassLoader(final Class<?> clazz) {
-        if (System.getSecurityManager() == null) {
-            return clazz.getClassLoader();
-        } else {
-            return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) clazz::getClassLoader);
-        }
+        return clazz.getClassLoader();
     }
 
+    @Deprecated
     static ClassLoader getContextClassLoader() {
-        if (System.getSecurityManager() == null) {
-            return Thread.currentThread().getContextClassLoader();
-        } else {
-            return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () ->
-                    Thread.currentThread().getContextClassLoader());
-        }
+        return Thread.currentThread().getContextClassLoader();
     }
 
+    @Deprecated
     static ClassLoader getSystemClassLoader() {
-        if (System.getSecurityManager() == null) {
-            return ClassLoader.getSystemClassLoader();
-        } else {
-            return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) ClassLoader::getSystemClassLoader);
-        }
+        return ClassLoader.getSystemClassLoader();
```

Comment:
We can inline these three into their usage sites. And honestly we don't need the versions in `PlatformDependent` either. Is this part of the later cleanup you're planning?

Reply:
Yes, exactly, I didn't want to introduce more noise in this PR - I will remove the deprecated methods from `PlatformDependent`, `PlatformDependent0` and `SocketUtils` in a follow up.